pontoon/checks/libraries/comparelocales.py

```diff
@@ -0,0 +1,166 @@
+from collections import (
+    namedtuple,
+)
+
+from compare_locales.checks import getChecker
+from compare_locales.parser import (
+    FluentParser,
+)
+
+CommentEntity = namedtuple(
+    'Comment', (
+        'all',
+    )
+)
+
+File = namedtuple(
+    'File', (
+        'file',
+        'locale',
+    )
+)
+
+# Because we can't pass the context to all entities passed to compare locales,
+# we have to create our equivalents of compare-locale's internal classes.
+ComparePropertiesEntity = namedtuple(
+    'ComparePropertiesEntity',
+    (
+        'key',
+        'val',
+        'raw_val',
+        'pre_comment',
+    )
+)
+CompareDTDEntity = namedtuple(
+    'CompareDTDEntity',
+    (
+        'key',
+        'val',
+        'raw_val',
```

Comment:
Silme does some escaping for DTD entries in https://github.com/mathjazz/silme/blob/f4dc4540334a2edfff885ffc1c35d123ab92dfbb/lib/silme/format/dtd/structure.py#L4-L8, which make compare-locales not run over the same string as in the pontoon db.
We also use html_escape in https://hg.mozilla.org/l10n/compare-locales/file/cdd7bc8c4232/compare_locales/parser.py#l14 to process `raw_val` to `val` in https://hg.mozilla.org/l10n/compare-locales/file/cdd7bc8c4232/compare_locales/parser.py#l336.

Reply:
ad1. @mathjazz from what I see, Pontoon don't call `process()` for Silme entities in order to escape (or substitute them with values) - is that correct?

Reply:
No, at least not directly. It could be called internally (I'd need to check).